### ISSUE TITLE
False positive for error E0277

### ISSUE THREAD

**Issue by Kaduo** [intellij-rust/intellij-rust#8786]
## Environment

* **IntelliJ Rust plugin version: 0.4.168.4555-221**
* **Rust toolchain version: 1.60**
* **IDE: CLion 2022.1**
* **Operating system: Arch Linux**

## Issue

The following code compiles without error but is linted incorrectly as erroneous:

```rust
trait MyTrait: From<Self::Foo> {
    type Foo;
}

impl<F> MyTrait for F {
    type Foo = F;
}

```

![image](https://user-images.githubusercontent.com/2149558/164978350-9ef83183-b296-4ea4-96f3-a210e6d64b75.png)

There are multiple open issues for false positives of E0277, but as far as I can tell none of them are directly related to this one. Feel free to close this issue if they are though.

I also wonder: is there any way I can delete this lint in the meantime?
